src/pipx/standalone_python.py

```diff
@@ -0,0 +1,182 @@
+import datetime
+import hashlib
+import json
+import platform
+import re
+import shutil
+import tarfile
+import urllib.error
+from functools import partial
+from typing import Any, Dict, List
+from urllib.request import urlopen
+
+from pipx.animate import animate
+from pipx.constants import PIPX_STANDALONE_PYTHON_CACHEDIR
+from pipx.util import PipxError
+
+# Much of the code in this module is adapted with extreme gratitude from
+# https://github.com/tusharsadhwani/yen/blob/main/src/yen/github.py
+
+MACHINE_SUFFIX: Dict[str, Dict[str, Any]] = {
+    "Darwin": {
+        "arm64": "aarch64-apple-darwin-install_only.tar.gz",
+        "x86_64": "x86_64-apple-darwin-install_only.tar.gz",
+    },
+    "Linux": {
+        "aarch64": {
+            "glibc": "aarch64-unknown-linux-gnu-install_only.tar.gz",
+            # musl doesn't exist
+        },
+        "x86_64": {
+            "glibc": "x86_64_v3-unknown-linux-gnu-install_only.tar.gz",
+            "musl": "x86_64_v3-unknown-linux-musl-install_only.tar.gz",
+        },
+    },
+    "Windows": {"AMD64": "x86_64-pc-windows-msvc-shared-install_only.tar.gz"},
+}
+
+GITHUB_API_URL = "https://api.github.com/repos/indygreg/python-build-standalone/releases/latest"
+PYTHON_VERSION_REGEX = re.compile(r"cpython-(\d+\.\d+\.\d+)")
+
+
+class NotAvailable(Exception):
+    """Raised when the asked Python version is not available."""
+
+
+def download_python_build_standalone(python_version: str):
+    """When all other python executable resolutions have failed,
+    attempt to download and use an appropriate python build
+    from https://github.com/indygreg/python-build-standalone
+    and unpack it into the pipx shared directory."""
+
+    # python_version can be a bare version number like "3.9" or a "binary name" like python3.10
+    # let's coerce it to a bare version number
```

Comment:
```suggestion
    # Split it to a bare version specifier
```

Didn't even know that word, `coerce`. Americans!

Reply:
hahaha I'm Canadian, but i get your meaning :) 

I've re-worded it